quint/src/cliCommands.ts

```diff
@@ -530,11 +530,7 @@ export async function runSimulator(prev: TypecheckedStage): Promise<CLIProcedure
   const mainEnd = prev.sourceMap.get(mainId)!.end!.index
   const result = compileAndRun(newIdGenerator(), mainText, mainStart, mainEnd, mainName, mainPath, options)
 
-  if (result.status === 'error') {
-    const newErrors = result.errors.map(mkErrorMessage(prev.sourceMap))
-    const errors = prev.errors ? prev.errors.concat(newErrors) : newErrors
-    return cliErr('run failed', { ...simulator, errors })
-  } else {
+  if (result.status !== 'error') {
```

Comment:
The logic all the way down to 579 is pretty hard to follow :sweat_smile: 

Due to the severity of this issue, I'm OK with merging your fix as is. But I suggest we do an immediate followup to refactor this logic so it is possible to follow, and would propose we use the approach in https://github.com/informalsystems/quint/blob/main/CONTRIBUTING.md#ensure-exhaustive-matches to take advantage of the type system to ensure we're handling all the possible statuses.

I'm happy to do that followup work if you've other things on your plate.